src/ServiceControl.Infrastructure.RavenDB/Extensions.cs

```diff
@@ -16,5 +18,40 @@ public static void Query<TState>(this DocumentDatabase db, string index, IndexQu
                 onItem(doc, state);
             }
         }
+
+        public static void ThrowWhenIndexErrors(this EmbeddableDocumentStore documentStore)
+        {
+            var statistics = documentStore.DatabaseCommands.GetStatistics();
+
+            if (statistics.Errors.Length > 0)
+            {
+                var text = new StringBuilder();
+                text.AppendLine("Detected RavenDB index errors, please start maintenance mode and resolve the following issues:");
+                foreach (var indexError in statistics.Errors)
+                {
+                    text.AppendLine($"Index [{indexError.IndexName}] error: {indexError.Error} (Action: {indexError.Action},  Doc: {indexError.Document}, At: {indexError.Timestamp})");
+                }
+                throw new Exception(text.ToString());
```

Comment:
Does this actually terminate the process or catch/display the error and continue? Maybe an `Environment.Exit(1)` is an alternative?

Reply:
@HEskandari Didn't test this yet but as this happens as a blocking call during bootstrap I assumed it would fail to start.